qiskit/pulse/channels/channel_store.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Store of channel registers.
+"""
+import logging
+from typing import List
+
+from qiskit.pulse import ChannelsError
+from .channel_register import ChannelRegister, AcquireChannelRegister, SnapshotChannelRegister
+from .output_channel import DriveChannel, ControlChannel, MeasureChannel
+from .output_channel_register import (DriveChannelRegister,
+                                      ControlChannelRegister,
+                                      MeasureChannelRegister)
+from .pulse_channel import AcquireChannel, SnapshotChannel
+
+logger = logging.getLogger(__name__)
+
+
+class ChannelStore:  # TODO: better name?
+    """Implement a channel store, which is a collection of channel registers.
+    It must be defined before constructing schedule."""
+
+    @classmethod
+    def create_from(cls, backend):
```

Comment:
I am uncertain if we should use a second class method constructor as above, or just have the ability to create the `ChannelStore` from just kwargs. See [the transpiler implementation](https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/transpiler/_transpiler.py) where either a backend or the required kwargs may be supplied and the init method knows how to handle both cases.

Reply:
Good suggestion. I'll think again about the interface of ChannelStore's constructor.